### PR TITLE
fix total calculation

### PR DESCRIPTION
**Pull request by n0script**
- since 0 as expression = false the total guess has been triggered
- also we can use data length if last page === first page

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/datorama/akita/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: akita/src/plugins/paginator/paginator-plugin.ts [hunk 1]
```diff
@@ -287,9 +287,14 @@ export class PaginatorPlugin<E> extends AkitaPlugin<E> {
 
         const { range, pagesControls } = this.config;
 
-        if (!this.pagination.total) {
-          response.total = response.perPage * response.lastPage;
-          this.pagination.total = response.total;
+        if (isNaN(Number.parseInt(this.pagination.total))) {
+          if (response.lastPage === 1) {
+            response.total = response.data ? response.data.length : 0;
+            this.pagination.total = response.total;
```
Comment (NetanelBasal) [inline akita/src/plugins/paginator/paginator-plugin.ts]: Call `this.pagination.total = response.total;` once.

File: akita/src/plugins/paginator/paginator-plugin.ts [hunk 2]
```diff
@@ -287,9 +287,14 @@ export class PaginatorPlugin<E> extends AkitaPlugin<E> {
 
         const { range, pagesControls } = this.config;
 
-        if (!this.pagination.total) {
-          response.total = response.perPage * response.lastPage;
-          this.pagination.total = response.total;
+        if (isNaN(Number.parseInt(this.pagination.total))) {
```
Comment (NetanelBasal) [inline akita/src/plugins/paginator/paginator-plugin.ts]: Omit the `Number.`.

File: akita/src/plugins/paginator/paginator-plugin.ts [hunk 3]
```diff
@@ -287,9 +287,15 @@ export class PaginatorPlugin<E> extends AkitaPlugin<E> {
 
         const { range, pagesControls } = this.config;
 
-        if (!this.pagination.total) {
-          response.total = response.perPage * response.lastPage;
-          this.pagination.total = response.total;
+        this.pagination.total = response.total;
```
Comment (NetanelBasal) [inline akita/src/plugins/paginator/paginator-plugin.ts]: I'm not sure what you did here. You need to call it once after you set the `response.total`.

Comment (n0script) [reply akita/src/plugins/paginator/paginator-plugin.ts]: sry for this misunderstanding

**Comment by NetanelBasal** [comment]
Please follow our git commit [convention](https://github.com/datorama/akita/blob/master/CONTRIBUTING.md#-commit-message-guidelines), so I will be able to merge it.

**Comment by NetanelBasal** [comment]
You can also add yourself as a contributor.

File: README.md [hunk 4]
```diff
@@ -33,6 +33,7 @@ Thanks goes to these wonderful people ([emoji key](https://github.com/kentcdodds
...
 | [<img src="https://avatars2.githubusercontent.com/u/14941988?v=4" width="100px;"/><br /><sub><b>Daniel Netzer</b></sub>](https://github.com/DanielNetzer)<br />[💻](https://github.com/NetanelBasal/akita/commits?author=DanielNetzer "Code") | [<img src="https://avatars3.githubusercontent.com/u/1046806?v=4" width="100px;"/><br /><sub><b>Emmanuel De Saint Steban</b></sub>](https://github.com/manudss)<br />[💻](https://github.com/NetanelBasal/akita/commits?author=manudss "Code") | [<img src="https://avatars1.githubusercontent.com/u/8677948?v=4" width="100px;"/><br /><sub><b>Tobias Wittwer</b></sub>](https://github.com/twittwer)<br />[💻](https://github.com/NetanelBasal/akita/commits?author=twittwer "Code") |
+[<img src="https://avatars3.githubusercontent.com/u/36441932?v=4" width="100px;"/><br /><sub><b>Miro Krenz</b></sub>](https://github.com/n0script)<br />[💻](https://github.com/n0script/akita/commits?author=n0script "Code") |
```
Comment (NetanelBasal) [inline README.md]: Cancel this and use `npm run contributors:add --yourgithubusername code`

**Comment by NetanelBasal** [comment]
It still not follows Akita's style guide. I did it and added you as a contributor. Next time, please read the style guide. Thanks.

**Pull request closed without merging**